### ISSUE TITLE
Adding a $key prevents component lifecycle `finally` clause

### ISSUE THREAD

**Issue by canadaduane** [bikeshaving/crank#267]
Today I encountered the slightly surprising behavior that adding a `$key` (or `crank-key`) attribute to a component causes the component's "finally" clause to no longer be called when unmounted:

```html
<!DOCTYPE html>
<script type="importmap">
  {
    "imports": {
      "@b9g/crank/dom": "https://cdn.jsdelivr.net/npm/@b9g/crank@0.5.4/dom/+esm",
      "@b9g/crank/standalone": "https://cdn.jsdelivr.net/npm/@b9g/crank@0.5.4/standalone/+esm"
    }
  }
</script>
<script type="module">
  import { createElement, jsx as html, Copy } from "@b9g/crank/standalone";
  import { renderer } from "@b9g/crank/dom";

  function* Box() {
    try {
      for (const {} of this) {
        yield html`
          <div style="border: 2px solid orange; border-radius: 5px">Hi!</div>
        `;
      }
    } finally {
      console.log("dismounted Box");
    }
  }

  function* App(message) {
    let show = true;

    setTimeout(() => {
      show = false;
      this.refresh();
    }, 2000);

    for (const {} of this) {
      yield show &&
        createElement(Box, {
          $key: "1",
          // $key: undefined,
        });
    }
  }

  renderer.render(createElement(App), document.body);
</script>
```

If you comment the `$key: "1"` line, the "dismounted Box" console log will appear as expected.

Without being too familiar with the internals of Crank.js, I assume there is some kind of memoization that has to take place when adding `$key`s, and this may be interacting with the component lifecycle, perhaps storing something in anticipation of the component moving to another location in the tree, and making it not unmount as usual?

**Comment by brainkim1** [comment]
Hey! Sorry for the tardy response! I am still trying to get access to my original account. The code which appears in the `finally` is run after the component is unmounted. In the example you gave, the component is never unmounted, so the `console.log()` statement wouldn’t run. You can call `renderer.render(null, document.body);` to see the unmount code execute.

**Comment by canadaduane** [comment]
> Hey! Sorry for the tardy response! I am still trying to get access to my original account.

No worries, thanks for weighing in!

> The code which appears in the `finally` is run after the component is unmounted. In the example you gave, the component is never unmounted, so the `console.log()` statement wouldn’t run. You can call `renderer.render(null, document.body);` to see the unmount code execute.

Ok, but shouldn't it be consistent, regardless of whether the `$key` is set?

**Comment by brainkim** [comment]
I HAVE ACCESS TO MY ACCOUNT!!! I AM INVESTIGATING! Sorry about the odd behavior!

**Comment by brainkim** [comment]
Fixed in v0.5.5
